### DISCUSSION TITLE
Migrating @root_validator(pre=True) pydantic > 2

### DISCUSSION THREAD

**Discussion by vkarisari** [pydantic/pydantic#9035] [Question]
Hi,

I have some code that initally used pydantic < 2. I have now upgraded to pydantic 2.6.0, and I get this warning: 

`PydanticDeprecatedSince20: Pydantic V1 style @root_validator validators are deprecated. You should migrate to Pydantic V2 style @model_validator validators, see the migration guide for more details. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.6/migration/`

My question is, should `@root_validator(pre=True)` be replaced with  `@model_validator(mode='before')`?

**Comment by Bishwas-py** [comment]
Yes, you can replace `@root_validator(pre=True)` with `@model_validator(mode='before')`.

**Reply by vkarisari** [reply]
Can or should? Does the `pre=True` corresponds to `mode='before'`?

**Reply by Bishwas-py** [reply]
I think you should. And IMO, `pre=True` resembles `mode='before'`.
